### ISSUE TITLE
Cannot run ozwdaemon docker image using older kernels, libQt5Core.so.5 not found

### ISSUE THREAD

**Issue by erikced** [OpenZWave/qt-openzwave#37]
When I try to start ozwdaemon build 47 (and others) all I get is
```
/usr/local/bin/ozwdaemon: error while loading shared libraries: libQt5Core.so.5: cannot open shared object file: No such file or directory
```

For some reason, all other libraries are found so it does not appear to be that the linker doesn't look in the right folders, and `libQt5Core.so.5` does exist:
```
$ docker run --rm --entrypoint=/usr/bin/ldd docker.io/openzwave/ozwdaemon:latest /usr/local/bin/ozwdaemon
        linux-vdso.so.1 (0x00007ffc9f7e7000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f5ae8104000)
        libqt-openzwave.so.1 => /usr/local/lib/libqt-openzwave.so.1 (0x00007f5ae7fed000)
        libqt-openzwavedatabase.so.1 => /usr/local/lib/libqt-openzwavedatabase.so.1 (0x00007f5ae7fe6000)
        libunwind.so.8 => /usr/lib/x86_64-linux-gnu/libunwind.so.8 (0x00007f5ae7fc9000)
        libQt5Mqtt.so.5 => /usr/local/lib/libQt5Mqtt.so.5 (0x00007f5ae7f99000)
        libQt5Network.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Network.so.5 (0x00007f5ae7dde000)
        libQt5Core.so.5 => not found
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f5ae7dbb000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f5ae7c37000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f5ae7a77000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f5ae81d4000)
        libopenzwave.so.1.6 => /usr/local/lib/libopenzwave.so.1.6 (0x00007f5ae78b5000)
        libQt5RemoteObjects.so.5 => /usr/lib/x86_64-linux-gnu/libQt5RemoteObjects.so.5 (0x00007f5ae7812000)
        libQt5Core.so.5 => not found
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f5ae77f6000)
        libQt5Core.so.5 => not found
        liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f5ae77ce000)
        libQt5Core.so.5 => not found
        libQt5Core.so.5 => not found
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f5ae75ae000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f5ae7469000)
        libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f5ae7451000)
        libQt5Core.so.5 => not found

$ docker run --rm --entrypoint=/bin/sh docker.io/openzwave/ozwdaemon:latest -c "/bin/ls -l /usr/lib/x86_64-linux-gnu/libQt*"
lrwxrwxrwx. 1 root root      20 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5Core.so.5 -> libQt5Core.so.5.12.5
lrwxrwxrwx. 1 root root      20 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5Core.so.5.12 -> libQt5Core.so.5.12.5
-rw-r--r--. 1 root root 5454088 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5Core.so.5.12.5
lrwxrwxrwx. 1 root root      20 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5DBus.so.5 -> libQt5DBus.so.5.12.5
lrwxrwxrwx. 1 root root      20 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5DBus.so.5.12 -> libQt5DBus.so.5.12.5
-rw-r--r--. 1 root root  619952 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5DBus.so.5.12.5
lrwxrwxrwx. 1 root root      23 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5Network.so.5 -> libQt5Network.so.5.12.5
lrwxrwxrwx. 1 root root      23 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5Network.so.5.12 -> libQt5Network.so.5.12.5
-rw-r--r--. 1 root root 1806136 Jan 30 21:03 /usr/lib/x86_64-linux-gnu/libQt5Network.so.5.12.5
lrwxrwxrwx. 1 root root      29 Dec 20 19:09 /usr/lib/x86_64-linux-gnu/libQt5RemoteObjects.so.5 -> libQt5RemoteObjects.so.5.12.5
lrwxrwxrwx. 1 root root      29 Dec 20 19:09 /usr/lib/x86_64-linux-gnu/libQt5RemoteObjects.so.5.12 -> libQt5RemoteObjects.so.5.12.5
-rw-r--r--. 1 root root  661680 Dec 20 19:09 /usr/lib/x86_64-linux-gnu/libQt5RemoteObjects.so.5.12.5
```

The host is running CentOS 7 / 7.7.1908, with the latest os-supplied docker (1.13.1 / 4ef4b30).

**Comment by erikced** [comment]
Based on https://github.com/Microsoft/WSL/issues/3023#issue-305053328 I tried stripping the ABI tag from `libQt5Core.so.5.12.5`, which appears to work but is probably not a great idea as QT 5.12 uses features only present in newer kernels (`renameat2`, `getrandom` syscalls)

Update:  From what I can tell it seems like RHEL/CentOS 7.7 kernels actually _do_ support `renameat2` and `getrandom`, which is not too surprising given the amount of backported functionality.

**Comment by Fishwaldo** [comment]
Thanks for investigating - it would have taken me a while to figure this out. I may explore switching to stretch instead of buster to see if that fixes it. Otherwise stripping that ABI compat section out maybe our only option.

**Comment by erikced** [comment]
So, just to follow up on this, the docker image works fine on CentOS 7.7 with a stripped qt lib. I don't think using stretch as base image will help as it was released with linux kernel 4.9.
I can create a patch to strip the lib when building the docker image if you want to, if so, should I also add a small startup script that checks if the relevant syscalls are available or should I skip it given that you'd have to go back to jessie or non-updated CentOS 7  i.e. relatively old distros to run containers on, to have any issues?

**Comment by Fishwaldo** [comment]
Sorry for late reply... working on lots of things at the moment. 

I looked around at other projects - i think we can unconditionally strip the binary of that section. 

Doing it in the builder image (not the final image) would be best, right after the compile 

Thanks!
